src/CommonDBTM.php

```diff
@@ -4330,7 +4330,10 @@ public function filterValues($display = true)
                         case 'email':
                         case 'string':
                         case 'itemlink':
-                            if (is_string($value) && ($length = mb_strlen($value, 'UTF-8')) > 255) {
+                            // Some 'completename' fields may be typed as 'itemlink' (e.g., in CommonTreeDropdown::class).
+                            // These should be excluded from the length check, as they are stored as TEXT or MEDIUMTEXT,
+                            // not VARCHAR. These fields are used solely as item links, not as standard string fields.
+                            if (!is_a(get_called_class(), CommonTreeDropdown::class, true) && is_string($value) && ($length = mb_strlen($value, 'UTF-8')) > 255) {
```

Comment:
If we choose to keep this filtering, we should specifically target the `completename` field, to minimize the impact of the change.

```suggestion
                            if ($key !== 'completename' && is_string($value) && ($length = mb_strlen($value, 'UTF-8')) > 255) {
```

Reply:
That was the purpose of my first comment. Maybe could it be applied on 10.0/bf branch (I guess issue already exists).